vscode-dotnet-runtime-library/src/Utils/FileUtilities.ts

```diff
@@ -12,172 +12,176 @@ import * as crypto from 'crypto';
-       }
+private innerWriteFile(scriptContent: string, filePath: string, eventStream? : IEventStream)
+{
+    scriptContent = eol.auto(scriptContent);
+    const existingScriptContent = fs.readFileSync(filePath).toString();
+    // fs.writeFile will replace the file if it exists.
+    // https://nodejs.org/api/fs.html#fswritefilefile-data-options-callback
+    if(scriptContent !== existingScriptContent)
+    {
+        fs.writeFileSync(filePath, scriptContent);
+        eventStream?.post(new DotnetFileWriteRequestEvent(`File content needed to be updated.`, new Date().toISOString(), filePath));
+    }
+    else
+    {
+        eventStream?.post(new DotnetFileWriteRequestEvent(`File content is an exact match, not writing file.`, new Date().toISOString(), filePath));
+    }
 
-       fs.chmodSync(filePath, 0o744);
-   }
+    fs.chmodSync(filePath, 0o744);
+}
 
-   /**
-    * @param directoryToWipe the directory to delete all of the files in if privilege to do so exists.
-    * @param fileExtensionsToDelete - if undefined, delete all files. if not, delete only files with extensions in this array in lower case.
-    */
-   public wipeDirectory(directoryToWipe : string, eventStream? : IEventStream, fileExtensionsToDelete? : string[])
-   {
-       if(!fs.existsSync(directoryToWipe))
-       {
-           eventStream?.post(new EmptyDirectoryToWipe(`The directory ${directoryToWipe} did not exist, so it was not wiped.`))
-           return;
-       }
+/**
+ * @param directoryToWipe the directory to delete all of the files in if privilege to do so exists.
+ * @param fileExtensionsToDelete - if undefined, delete all files. if not, delete only files with extensions in this array in lower case.
+ */
+public wipeDirectory(directoryToWipe : string, eventStream? : IEventStream, fileExtensionsToDelete? : string[])
+{
+    if(!fs.existsSync(directoryToWipe))
+    {
+        eventStream?.post(new EmptyDirectoryToWipe(`The directory ${directoryToWipe} did not exist, so it was not wiped.`))
+        return;
+    }
 
-       // Use rimraf to delete all of the items in a directory without the directory itself.
-       fs.readdirSync(directoryToWipe).forEach(f =>
-       {
-           try
-           {
-               eventStream?.post(new FileToWipe(`The file ${f} is being deleted.`))
-               if(!fileExtensionsToDelete || path.extname(f).toLocaleLowerCase() in fileExtensionsToDelete)
-               fs.rmSync(path.join(directoryToWipe, f));
-           }
-           catch(error : any)
-           {
-               eventStream?.post(new SuppressedAcquisitionError(error, `Failed to delete ${f} when marked for deletion.`));
-           }
-       });
-   }
+        // Use rimraf to delete all of the items in a directory without the directory itself.
+        fs.readdirSync(directoryToWipe).forEach(f =>
+        {
+            try
+            {
+                eventStream?.post(new FileToWipe(`The file ${f} may be deleted.`))
```

Comment:
Since we know two lines down whether f is going to be deleted, can we just wait until we know for sure?

Reply:
This is for the log to show whether it detected the file or not. If it doesnt try to delete the file it wont show up in the log